pkg/registry/apps/alerting/rules/alertrule/compat.go

```diff
@@ -0,0 +1,361 @@
+	}
+
+	if rule.For != 0 {
+		k8sRule.Spec.For = util.Pointer(rule.For.String())
+	}
+
+	if rule.KeepFiringFor != 0 {
+		k8sRule.Spec.KeepFiringFor = util.Pointer(rule.KeepFiringFor.String())
+	}
+
+	if rule.PanelID != nil && rule.DashboardUID != nil &&
+		*rule.PanelID > 0 && *rule.DashboardUID != "" {
+		k8sRule.Spec.PanelRef = &model.AlertRuleV0alpha1SpecPanelRef{
+			PanelID:      *rule.PanelID,
+			DashboardUID: *rule.DashboardUID,
+		}
+	}
+
+	for k, v := range rule.Annotations {
+		k8sRule.Spec.Annotations[k] = model.AlertRuleTemplateString(v)
+	}
+
+	for k, v := range rule.Labels {
+		k8sRule.Spec.Labels[k] = model.AlertRuleTemplateString(v)
+	}
+
+	for _, query := range rule.Data {
+		k8sQuery := model.AlertRuleQuery{
+			QueryType:     query.QueryType,
+			Model:         query.Model,
+			DatasourceUID: model.AlertRuleDatasourceUID(query.DatasourceUID),
+			Source:        util.Pointer(rule.Condition == query.RefID),
+		}
+		if time.Duration(query.RelativeTimeRange.From) > 0 || time.Duration(query.RelativeTimeRange.To) > 0 {
+			k8sQuery.RelativeTimeRange = &model.AlertRuleRelativeTimeRange{
+				From: model.AlertRulePromDurationWMillis(query.RelativeTimeRange.From.String()),
+				To:   model.AlertRulePromDurationWMillis(query.RelativeTimeRange.To.String()),
+			}
+		}
+		k8sRule.Spec.Data[query.RefID] = k8sQuery
+	}
+
+	for _, setting := range rule.NotificationSettings {
+		nfSetting := model.AlertRuleV0alpha1SpecNotificationSettings{
+			Receiver: setting.Receiver,
+			GroupBy:  setting.GroupBy,
+		}
+		if setting.GroupWait != nil {
+			nfSetting.GroupWait = util.Pointer(setting.GroupWait.String())
+		}
+		if setting.GroupInterval != nil {
+			nfSetting.GroupInterval = util.Pointer(setting.GroupInterval.String())
+		}
+		if setting.RepeatInterval != nil {
+			nfSetting.RepeatInterval = util.Pointer(setting.RepeatInterval.String())
+		}
+		if setting.MuteTimeIntervals != nil {
+			nfSetting.MuteTimeIntervals = make([]model.AlertRuleMuteTimeIntervalRef, 0, len(setting.MuteTimeIntervals))
+			for _, m := range setting.MuteTimeIntervals {
+				nfSetting.MuteTimeIntervals = append(nfSetting.MuteTimeIntervals, model.AlertRuleMuteTimeIntervalRef(m))
+			}
+		}
+		if setting.ActiveTimeIntervals != nil {
+			nfSetting.ActiveTimeIntervals = make([]model.AlertRuleActiveTimeIntervalRef, 0, len(setting.ActiveTimeIntervals))
+			for _, a := range setting.ActiveTimeIntervals {
+				nfSetting.ActiveTimeIntervals = append(nfSetting.ActiveTimeIntervals, model.AlertRuleActiveTimeIntervalRef(a))
+			}
+		}
+		k8sRule.Spec.NotificationSettings = &nfSetting
+	}
+
+	meta, err := utils.MetaAccessor(k8sRule)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get metadata: %w", err)
+	}
+	meta.SetFolder(rule.NamespaceUID)
+	if rule.UpdatedBy != nil {
+		meta.SetUpdatedBy(string(*rule.UpdatedBy))
+		k8sRule.SetUpdatedBy(string(*rule.UpdatedBy))
+	}
+	meta.SetUpdatedTimestamp(&rule.Updated)
+	k8sRule.SetUpdateTimestamp(rule.Updated)
```

Comment:
Why do we need two annotations dedicated to updated timestamp?
It looks like 
```
      "grafana.app/updatedTimestamp": "2025-09-05T20:01:23Z",
      "grafana.com/updateTimestamp": "2025-09-05T16:01:23-04:00",
```

Reply:
I don't really know, but they both existed and I didn't know which one to use.